releaseNotes/Microsoft.Featuremanagement.Telemetry.ApplicationInsights.md

```diff
@@ -2,6 +2,21 @@
 
 [Source code][source_code] | [Package (NuGet)][package] | [Samples][samples] | [Product documentation][docs]
 
+## 4.0.0 - September 24, 2024
+
+### Enhancements
+
+#### Application Insights
+
+```csharp
+builder.Services.AddFeatureManagement()
+    .AddApplicationInsightsTelemetry();
+```
+
+To export telemetry to Application Insights as custom events, we offer a new builder method `IFeatureManagementBuilder.AddApplicationInsightsTelemetry()`. This causes outgoing Application Insights telemetry to be tagged with TargetingId and will emit evaluation events to the custom events table- using the `TelemetryClient` if available in DI.
```

Comment:
Too much implementation detail. Users should read documents for that level of detail.

#### Application Insights

Added support for Application Insights telemetry. To publish feature flag evaluation data to Application Insights, register the Application Insights telemetry publisher as shown below.

```csharp
builder.Services.AddFeatureManagement()
       .AddApplicationInsightsTelemetry();
```

Reply:
We can simplify it if you prefer. But I think we need to call out that all outgoing events will have extra data appended to them. It's not only the feature evaluation data. (aka- if someone uses this but never calls `IsEnabledAsync()/GetVariantAsync()`- there's still a change)

Reply:
The TargetingId is not even added by the AddApplicationInsightsTelemetry API, is it? Let users read documentation.

Reply:
It is. Calling `.AddApplicationInsightsTelemetry()` adds the `TargetingTelemetryInitializer`. This appends `TargetingId` onto all outgoing events as long as `TargetingId` is available from the `ITargetingContextAccessor`.

How about we just adjust your version to say:
> Added support for Application Insights telemetry. To publish feature flag evaluation data and tag outgoing events with targeting information, register the Application Insights telemetry publisher as shown below.